### ISSUE TITLE
Training Loop Issue:    ForwardModelsTrain()

### ISSUE THREAD

**Issue by moutaigua8183** [facebookresearch/vilbert-multi-task#30]
Doesn't anybody feel it probmatic or twisted about the training loop?
In ForwardModelsTrain(), there is a loop over dataloader to use the batch data one after another.  But there is a "return loss, batch_score" within the loop, meaning that the function only uses the first batch before exiting.   After the loss and score of the batch are returned,  the parent function does the back propagration and a few other tasks as it is supposed to do.      

My question is that  should the loop over the dataloader outside the ForwardModelsTrain(), replacing the line `for step in range(median_num_iter):`   ?

**Comment by moutaigua8183** [comment]
never mind.  The original implementation should be correct.   I am going to close this.
